### ISSUE TITLE
Weapon Camos using a 2nd UV channel

### ISSUE THREAD

**Issue by RickGreeve** [Facepunch/sbox-issues#5810]
### For?

S&Box

### What can't you do?

early on during the development of weapons I spoke with @trwb about guns having a 2nd UV channel for camos.
Call of Duty does this a lot (not sure about current day, but I remember making 2nd UV channels on guns for CoD WWII).
Could be a great addition for customization in HC1 and any other similar first person shooter in the future.

I understand if something like this is at the back of the priority list, but if there is time it could be worth a discussion. 
And of course I'm happy to provide any art related assets needed (2nd UV channels, camos etc.) and work together on anything else 

### How would you like it to work?

the shader would need a couple things:

1. main thing is being able to access the 2nd UV channel which I believe Compex already does. 
2. a mask input with a Black and White or RGB map to mask of where the camo actually goes (RGB would allow us to have multiple wear states like CS has for example altough I believe they handle it differently).
3. a input for camo's themselves (color, roughness and metalness). 
4. maybe an input for how many camo options so we can have multiple options on the same gun.

### What have you tried?

not a lot, enabling secondary Uv in the shader doesn't have many options besides AO

**Comment by RickGreeve** [comment]
I am realising that making a 2nd UV channel for every material on the Viewmodel and every Worldmodel and LOD would be way to much manual labour. Thankfull @bakscratch pointed out to me that we have triplanar projection working so we can use that instead. Then it's just a matter of making the masks and some camo options and we're good to go without a second UV channel as triplanar does everything I would require of a 2nd UV channel. For more complex skins I can make something bespoke instead and use material sets instead so that works out too. 

I'll see if I make something of decent quality with what Louie provided me with sofar.

**Comment by bakscratch** [comment]
https://github.com/user-attachments/assets/43ed1973-85e1-43d5-aa6b-96d84b9caeed

Some of the progress

**Comment by bakscratch** [comment]
![image](https://github.com/user-attachments/assets/0b20b618-2689-48fa-9ae3-64b953f0dec3)

Pushed with a test material

**Comment by RickGreeve** [comment]
@trwb We have been making some progress with this over hackweek

using a tileble made in Designer (screenshots are from sbox)
![image (7)](https://github.com/user-attachments/assets/bcbca9ec-279f-4227-bae5-53906ee79f24)

using tintable tilebles 
![image (9)](https://github.com/user-attachments/assets/03134465-d8c7-4528-aa62-3c3ee9e55404)
![image (8)](https://github.com/user-attachments/assets/6752cb12-8018-4a0c-9fb5-3fa19292630d)

graph and the first tiling texture from above 
![weapon_camo_spraynet_color](https://github.com/user-attachments/assets/1ad27b50-1b15-4bb3-8005-90b7b7606749)
![image (10)](https://github.com/user-attachments/assets/7eb8dbf1-7b50-4032-84f5-309ddc9996a8)

I don't know if I can find the time to work on this much after hackweek. I think documentation and finishing the LODs for the sniper and scope take priority and then I really should finish off the rest of those grenades we need in Nicked I think, by that point its probably a good idea to start working on the missile launcher before doing camo's.

But some good progress made over hackweek, there are some kinks i'm working out with louie but I am confident we are getting some good results with this workflow!

**Comment by trwb** [comment]
Sounds good to me. Is there plans to refine the scratch mask for the camos? Have you tried some softer falloff in some areas so it doesn't look such a sharp transition?

**Comment by RickGreeve** [comment]
@trwb 
Yeah still a lot of back and forth on the mask front, I'll make sure to ask for feedback once I am at a good point with the entire gun and post ref as well, but that might be a while from now
